scipy/optimize/_zeros_py.py

```diff
@@ -1392,3 +1393,344 @@ def toms748(f, a, b, args=(), k=1,
+            ``-4`` : Iteration was terminated by `callback`.
+            ``1`` : The algorithm is proceeding normally (in `callback` only).
+        success : bool
+            ``True`` when the algorithm terminated successfully (status ``0``).
+        fun : float
+            The value of `func` evaluated at `x`.
+        xl, xr : float
+            The lower and upper ends of the bracket.
+        fl, fr : float
+            The function value at the lower and upper ends of the bracket.
+
+    Notes
+    -----
+    Implemented based on Chandrupatla's original paper [1]_.
+
+    If ``xl`` and ``xr`` are the left and right ends of the bracket,
+    ``xmin = xl if abs(func(xl)) <= abs(func(xr)) else xr``,
+    and ``fmin0 = min(func(a), func(b))``, then the algorithm is considered to
+    have converged when ``abs(xr - xl) < xatol + abs(xmin) * xrtol`` or
+    ``fun(xmin) <= fatol + abs(fmin0) * frtol``. This is equivalent to the
+    termination condition described in [1]_ with ``xrtol = 4e-10``,
+    ``xatol = 1e-5``, and ``fatol = frtol = 0``. The default values are
+    ``xatol = 2e-12``, ``xrtol = 4 * np.finfo(float).eps``, ``frtol = 0``,
+    and ``fatol`` is the smallest normal number of the ``dtype`` returned
+    by ``func``.
+
+    References
+    ----------
+
+    .. [1] Chandrupatla, Tirupathi R.
+        "A new hybrid quadratic/bisection algorithm for finding the zero of a
+        nonlinear function without using derivatives".
+        Advances in Engineering Software, 28(3), 145-149.
+        https://doi.org/10.1016/s0965-9978(96)00051-8
+
+    See Also
+    --------
+    brentq, brenth, ridder, bisect, newton
+
+    Examples
+    --------
+    >>> from scipy import optimize
+    >>> def f(x, c):
+    ...     return x**3 - 2*x - c
+    >>> c = 5
+    >>> res = optimize._zeros_py._chandrupatla(f, 0, 3, args=(c,))
+    >>> res.x
+    2.0945514818937463
+
+    >>> c = [3, 4, 5]
+    >>> res = optimize._zeros_py._chandrupatla(f, 0, 3, args=(c,))
+    >>> res.x
+    array([1.8932892 , 2.        , 2.09455148])
+
+    """
+
+    res = _chandrupatla_iv(func, a, b, args, xatol, xrtol,
+                           fatol, frtol, maxiter, callback)
+    func, a, b, args, xatol, xrtol, fatol, frtol, maxiter, callback = res
+
+    # Initialization
+    x1, f1, x2, f2, shape, dtype = _initialize_xf(func, a, b, args)
+    status = np.full_like(x1, _EINPROGRESS, dtype=int)  # in progress
+    active = np.arange(x1.size)  # indices of in-progress elements
+    nit, nfev = 0, 2  # two function evaluations performed above
+    cb_terminate = False
+    fatol = fatol or np.finfo(dtype).smallest_normal
+    frtol = frtol * np.minimum(np.abs(f1), np.abs(f2))
+    tols = dict(xatol=xatol, xrtol=xrtol, fatol=fatol, frtol=frtol)
+    # Elements of `x1`, `f1`, etc., are stored in this `OptimizeResult`
+    # once a termination condition is met, and then the arrays are condensed
+    # to reduce unnecessary computation.
```

Comment:
This is important. `_check_termination` looks for elements that don't require further iteration, whether the algorithm has converged or there is a problem (invalid bracket, NaNs, or infs). These elements are removed from the like `x1`, `f1`, etc, arrays ("compression") and stored in the appropriate places of the `OptimizeResult`.